### ISSUE TITLE
RDS TF Plans throw EOF error

### ISSUE THREAD

**Issue by rjmii** [terraform-providers/terraform-provider-aws#2112]
### Terraform Version
Terraform v0.10.8

### AWS Provider Version
0.1.4_x4

```hcl
### Terraform Configuration Files
##main.tf module
#--------------------------------------------------------------
# This module creates an rds instance from a specified db snapshot
#--------------------------------------------------------------

resource "aws_db_instance" "db_instance" {
  engine                  = "mysql"
  engine_version          = "${var.engine_version}"
  instance_class          = "${var.instance_class}"
  db_subnet_group_name    = "${var.db_subnet_group_name}"
  parameter_group_name    = "${var.parameter_group_name}"
  snapshot_identifier     = "${var.snapshot_identifier}"
  identifier              = "${var.identifier}"
  skip_final_snapshot     = "${var.skip_final_snapshot}"
  apply_immediately       = "${var.apply_immediately}"
  backup_retention_period = "${var.backup_retention_period}"
  replicate_source_db     = "${var.replicate_source_db}"

  tags {
    Manager = "Terraform"
    YLASTIC = ""
  }
}
```

```hcl
##main.tf
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# Deploy a RDS MySQL Instance
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

#This creates the shared state tracking and locking for terraform. It can not be used without performing terraform init.
terraform {
  backend "s3" {
    bucket         = "newton-tf-state"
    key            = "newton/awsdev/db"
    profile        = "newton-dev"
    region         = "us-east-1"
    dynamodb_table = "newton-tf-state-lock"
  }
}

provider "aws" {
  region  = "${var.aws_region}"
  profile = "${var.aws_profile}"
  version = "~> 0.1.3"
}

provider "template" {
  version = "~> 0.1.1"
}

#grabs the latest db snapshot from the identified instance
data "aws_db_snapshot" "db_snapshot" {
  most_recent            = true
  db_instance_identifier = "devdb-refreshed-from-pro-nightly"
}

module "db_main" {
  source = "../../../../../../../../rjmii/terraform_modules/modules/db_instance"

  #instance info
  snapshot_identifier     = "${data.aws_db_snapshot.db_snapshot.id}"
  identifier              = "devdb-awsdev"
  apply_immediately       = "true"
  backup_retention_period = 1
  parameter_group_name    = "dev-db-params-56-master"
  db_subnet_group_name    = "dev-db-subnets"
}

module "db_replica01" {
  source = "../../../../../../../../rjmii/terraform_modules/modules/db_instance"

  #instance info
  identifier           = "devdb-awsdev-replica01"
  replicate_source_db  = "${module.db_main.identifier}"
  instance_class       = "db.m4.large"
  apply_immediately    = "true"
  parameter_group_name = "dev-db-params-56-slave01"
}

module "db_replica02" {
  source = "../../../../../../../../rjmii/terraform_modules/modules/db_instance"

  #instance info
  identifier           = "devdb-awsdev-replica02"
  replicate_source_db  = "${module.db_main.identifier}"
  instance_class       = "db.m4.large"
  apply_immediately    = "true"
  parameter_group_name = "dev-db-params-56-slave02"
}
```

### Affected Resource(s)

RDS Instances

### Debug Output
https://gist.github.com/rjmii/f5d02bd579e705996aea5fccd093323f

### Expected Behavior
For plan, destroy, or apply terraform should execute the the plan and output the results 

### Actual Behavior
Terraform displays this

```
Acquiring state lock. This may take a few moments...
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.aws_db_snapshot.db_snapshot: Refreshing state...
Releasing state lock. This may take a few moments...

Error: Error refreshing state: 1 error(s) occurred:

* data.aws_db_snapshot.db_snapshot: 1 error(s) occurred:

* data.aws_db_snapshot.db_snapshot: data.aws_db_snapshot.db_snapshot: unexpected EOF
```

### Steps to Reproduce
1. `terraform plan` or `terraform destory` or `terraform apply`

### Important Factoids
None

**Comment by rjmii** [comment]
We got it working by changing the version from 0.1 (in this case 0.1.3) to 1.0 . We did it for both provider and template. 

Still something is wonky with the latest 0.1 version number.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.1](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#171-january-19-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
